src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -48,42 +48,90 @@ class java_bytecode_convert_classt:public messaget
   {
   }
 
-  void operator()(const java_bytecode_parse_treet &parse_tree)
+  /// Converts a class parse tree into a class symbol and adds it to the
+  /// symbol table.
+  /// \param parse_trees: The parse trees found for the class to be converted.
+  /// \remarks
+  ///   Allows multiple definitions of the same class to appear on the
+  ///   classpath, so long as all but the first definition are marked with the
```

Comment:
This implies if the first is annotated with `OverlayClassImplementation` this won't work, is that correct/desirable (it seems like if exactly one are marked with the `OverlayClassImplementation` tag as the ordering of loading classes doesn't seem like an ideal thing to depend on.

Reply:
You should definitely put your overlay classes in a folder/jar that you add to the end of your classpath. We depend on the order of the classpath already as we currently only load the first instance seen.